pkg/core/chain/chain.go

```diff
@@ -336,11 +337,33 @@ func (c *Chain) ProcessSyncTimerExpired(strPeerAddr string) error {
 	return nil
 }
 
+func throttle(startTimeMilli int64, maxDelayMilli int64) (time.Duration, error) {
+	if startTimeMilli == 0 || maxDelayMilli == 0 {
+		return time.Duration(0), errors.New("invalid input")
+	}
+
+	delta := time.Now().UnixMilli() - startTimeMilli
+
+	if delta > 0 && delta < maxDelayMilli {
+		duration := time.Duration((maxDelayMilli - delta) * int64(time.Millisecond))
+
+		if duration > 0 {
+			return duration, nil
+		}
+	}
+
+	return time.Duration(0), errors.New("max delay exceeded")
+}
+
 // acceptSuccessiveBlock will accept a block which directly follows the chain
 // tip, and advertises it to the node's peers.
 func (c *Chain) acceptSuccessiveBlock(blk block.Block, kadcastHeight byte) error {
+	startTime := time.Now().UnixMilli()
+
 	log.WithField("height", blk.Header.Height).Trace("accepting succeeding block")
 
+	prevBlockTimestamp := c.tip.Header.Timestamp
```

Comment:
I'm a bit concerned to use timestamps in environment where no ntp is configured

Could the received instant be used for this scenario?

Reply:
Timestamp in block header is a field being validated. It's guaranteed that it's bigger than prevBlock timestamp.

Reply:
Ok, in this case I open this followUp issue to tackle some odd corner cases

https://github.com/dusk-network/dusk-blockchain/issues/1334